app/services/FileTypeCheckingService.scala

```diff
@@ -31,19 +31,30 @@ class FileTypeCheckingService @Inject()(fileTypeDetector: FileTypeDetector, serv
     location: S3ObjectLocation,
     objectContent: ObjectContent,
     objectMetadata: ObjectMetadata): Future[FileCheckingResult] = {
-    val fileType = fileTypeDetector.detectType(objectContent.inputStream, objectMetadata.originalFilename)
+    val fileType         = fileTypeDetector.detectType(objectContent.inputStream, objectMetadata.originalFilename)
```

Comment:
method is not async, does the return type need to be a future?

Reply:
This is to do with how we chain file scans, to provide a consistent way of handling multiple checks that a file needs to pass. I've added in a Trait that makes this relationship explicit.